src/arbitraries/MappedArbitrary.ts

```diff
@@ -32,8 +39,17 @@ export class MappedArbitrary<A, B> extends Arbitrary<B> {
   }
 
   canGenerate(pick: FluentPick<B>) {
-    /* && pick.value === this.f(pick.original) */
-    return this.baseArbitrary.canGenerate({value: pick.original, original: pick.original})
+    const inverseValues = !(this.config && this.config.inverseFunction) ? [pick.original] :
+      (Array.isArray(this.config.inverseFunction(pick.value)) ?
```

Comment:
Why are you calculating the `inverseFunction` twice?

Reply:
I was doing that just to keep the code clean and because it was an easy way to achieve what I wanted. However, I decided to remove the return-type polymorphism (as suggested by @ruippeixotog), which turned out to remove the code that you mentioned here.